File: server-scala-play-server/app/api/ApplicationConfigApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.ApplicationConfigResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class ApplicationConfigApiController @Inject()(cc: ControllerComponents, api: ApplicationConfigApi) extends AbstractController(cc) {
  /**
    * POST /api/3.18/appconfig/create?accountId=[value]&appKey=[value]&configVersion=[value]&assetId=[value]&retailerId=[value]&retailerLocationId=[value]&udid=[value]
    */
  def createApplicationConfig(): Action[AnyContent] = Action { request =>
    def executeApi(): ApplicationConfigResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      val configVersion = request.getQueryString("configVersion")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("configVersion", "query string")
        }
        
      val assetId = request.getQueryString("assetId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assetId", "query string")
        }
        
      val retailerId = request.getQueryString("retailerId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val udid = request.getQueryString("udid")
        
      api.createApplicationConfig(accountId, appKey, configVersion, assetId, retailerId, retailerLocationId, udid)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/appconfig/delete?accountId=[value]&configId=[value]
    */
  def deleteApplicationConfig(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val configId = request.getQueryString("configId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("configId", "query string")
        }
        
      api.deleteApplicationConfig(accountId, configId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/appconfig/get?accountId=[value]&configId=[value]
    */
  def getApplicationConfig(): Action[AnyContent] = Action { request =>
    def executeApi(): ApplicationConfigResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val configId = request.getQueryString("configId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("configId", "query string")
        }
        
      api.getApplicationConfig(accountId, configId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/appconfig/getbyversion?appKey=[value]&configVersion=[value]&retailerId=[value]&retailerLocationId=[value]&udid=[value]&allowOlderVersions=[value]
    */
  def getApplicationConfigByConfigVersion(): Action[AnyContent] = Action { request =>
    def executeApi(): ApplicationConfigResponse = {
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      val configVersion = request.getQueryString("configVersion")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("configVersion", "query string")
        }
        
      val retailerId = request.getQueryString("retailerId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val udid = request.getQueryString("udid")
        
      val allowOlderVersions = request.getQueryString("allowOlderVersions")
        .map(value => value.toBoolean)
        
      api.getApplicationConfigByConfigVersion(appKey, configVersion, retailerId, retailerLocationId, udid, allowOlderVersions)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/appconfig/search?accountId=[value]&appKey=[value]&retailerId=[value]&retailerLocationId=[value]&udid=[value]&configVersion=[value]&sortField=[value]&descending=[value]&start=[value]&limit=[value]
    */
  def searchApplicationConfig(): Action[AnyContent] = Action { request =>
    def executeApi(): List[ApplicationConfigResponse] = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        
      val retailerId = request.getQueryString("retailerId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val udid = request.getQueryString("udid")
        
      val configVersion = request.getQueryString("configVersion")
        
      val sortField = request.getQueryString("sortField")
        
      val descending = request.getQueryString("descending")
        .map(value => value.toBoolean)
        
      val start = request.getQueryString("start")
        .map(value => value.toInt)
        
      val limit = request.getQueryString("limit")
        .map(value => value.toInt)
        
      api.searchApplicationConfig(accountId, appKey, retailerId, retailerLocationId, udid, configVersion, sortField, descending, start, limit)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/appconfig/update?accountId=[value]&configId=[value]&appKey=[value]&configVersion=[value]&assetId=[value]&retailerId=[value]&retailerLocationId=[value]&udid=[value]
    */
  def updateApplicationConfig(): Action[AnyContent] = Action { request =>
    def executeApi(): ApplicationConfigResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val configId = request.getQueryString("configId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("configId", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        
      val configVersion = request.getQueryString("configVersion")
        
      val assetId = request.getQueryString("assetId")
        .map(value => value.toLong)
        
      val retailerId = request.getQueryString("retailerId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val udid = request.getQueryString("udid")
        
      api.updateApplicationConfig(accountId, configId, appKey, configVersion, assetId, retailerId, retailerLocationId, udid)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/ApplicationConfigApiController.java
package controllers;

import apimodels.ApplicationConfigResponse;
import java.math.BigDecimal;
import apimodels.SirqulResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class ApplicationConfigApiController extends Controller {
    private final ApplicationConfigApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private ApplicationConfigApiController(Config configuration, ApplicationConfigApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result createApplicationConfig(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        String valueconfigVersion = request.getQueryString("configVersion");
        String configVersion;
        if (valueconfigVersion != null) {
            configVersion = valueconfigVersion;
        } else {
            throw new IllegalArgumentException("'configVersion' parameter is required");
        }
        String valueassetId = request.getQueryString("assetId");
        Long assetId;
        if (valueassetId != null) {
            assetId = Long.parseLong(valueassetId);
        } else {
            throw new IllegalArgumentException("'assetId' parameter is required");
        }
        String valueretailerId = request.getQueryString("retailerId");
        Long retailerId;
        if (valueretailerId != null) {
            retailerId = Long.parseLong(valueretailerId);
        } else {
            retailerId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valueudid = request.getQueryString("udid");
        String udid;
        if (valueudid != null) {
            udid = valueudid;
        } else {
            udid = null;
        }
        return imp.createApplicationConfigHttp(request, version, accountId, appKey, configVersion, assetId, retailerId, retailerLocationId, udid);
    }

    @ApiAction
    public Result deleteApplicationConfig(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueconfigId = request.getQueryString("configId");
        Long configId;
        if (valueconfigId != null) {
            configId = Long.parseLong(valueconfigId);
        } else {
            throw new IllegalArgumentException("'configId' parameter is required");
        }
        return imp.deleteApplicationConfigHttp(request, version, accountId, configId);
    }

    @ApiAction
    public Result getApplicationConfig(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueconfigId = request.getQueryString("configId");
        Long configId;
        if (valueconfigId != null) {
            configId = Long.parseLong(valueconfigId);
        } else {
            throw new IllegalArgumentException("'configId' parameter is required");
        }
        return imp.getApplicationConfigHttp(request, version, accountId, configId);
    }

    @ApiAction
    public Result getApplicationConfigByConfigVersion(Http.Request request, BigDecimal version) throws Exception {
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        String valueconfigVersion = request.getQueryString("configVersion");
        String configVersion;
        if (valueconfigVersion != null) {
            configVersion = valueconfigVersion;
        } else {
            throw new IllegalArgumentException("'configVersion' parameter is required");
        }
        String valueretailerId = request.getQueryString("retailerId");
        Long retailerId;
        if (valueretailerId != null) {
            retailerId = Long.parseLong(valueretailerId);
        } else {
            retailerId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valueudid = request.getQueryString("udid");
        String udid;
        if (valueudid != null) {
            udid = valueudid;
        } else {
            udid = null;
        }
        String valueallowOlderVersions = request.getQueryString("allowOlderVersions");
        Boolean allowOlderVersions;
        if (valueallowOlderVersions != null) {
            allowOlderVersions = Boolean.valueOf(valueallowOlderVersions);
        } else {
            allowOlderVersions = false;
        }
        return imp.getApplicationConfigByConfigVersionHttp(request, version, appKey, configVersion, retailerId, retailerLocationId, udid, allowOlderVersions);
    }

    @ApiAction
    public Result searchApplicationConfig(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        String valueretailerId = request.getQueryString("retailerId");
        Long retailerId;
        if (valueretailerId != null) {
            retailerId = Long.parseLong(valueretailerId);
        } else {
            retailerId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valueudid = request.getQueryString("udid");
        String udid;
        if (valueudid != null) {
            udid = valueudid;
        } else {
            udid = null;
        }
        String valueconfigVersion = request.getQueryString("configVersion");
        String configVersion;
        if (valueconfigVersion != null) {
            configVersion = valueconfigVersion;
        } else {
            configVersion = null;
        }
        String valuesortField = request.getQueryString("sortField");
        String sortField;
        if (valuesortField != null) {
            sortField = valuesortField;
        } else {
            sortField = "CONFIG_VERSION_INDEX";
        }
        String valuedescending = request.getQueryString("descending");
        Boolean descending;
        if (valuedescending != null) {
            descending = Boolean.valueOf(valuedescending);
        } else {
            descending = true;
        }
        String valuestart = request.getQueryString("start");
        Integer start;
        if (valuestart != null) {
            start = Integer.parseInt(valuestart);
        } else {
            start = 0;
        }
        String valuelimit = request.getQueryString("limit");
        Integer limit;
        if (valuelimit != null) {
            limit = Integer.parseInt(valuelimit);
        } else {
            limit = 20;
        }
        return imp.searchApplicationConfigHttp(request, version, accountId, appKey, retailerId, retailerLocationId, udid, configVersion, sortField, descending, start, limit);
    }

    @ApiAction
    public Result updateApplicationConfig(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueconfigId = request.getQueryString("configId");
        Long configId;
        if (valueconfigId != null) {
            configId = Long.parseLong(valueconfigId);
        } else {
            throw new IllegalArgumentException("'configId' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        String valueconfigVersion = request.getQueryString("configVersion");
        String configVersion;
        if (valueconfigVersion != null) {
            configVersion = valueconfigVersion;
        } else {
            configVersion = null;
        }
        String valueassetId = request.getQueryString("assetId");
        Long assetId;
        if (valueassetId != null) {
            assetId = Long.parseLong(valueassetId);
        } else {
            assetId = null;
        }
        String valueretailerId = request.getQueryString("retailerId");
        Long retailerId;
        if (valueretailerId != null) {
            retailerId = Long.parseLong(valueretailerId);
        } else {
            retailerId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valueudid = request.getQueryString("udid");
        String udid;
        if (valueudid != null) {
            udid = valueudid;
        } else {
            udid = null;
        }
        return imp.updateApplicationConfigHttp(request, version, accountId, configId, appKey, configVersion, assetId, retailerId, retailerLocationId, udid);
    }

}
